### ISSUE TITLE
InvalidOperationException in usage reporter

### ISSUE THREAD

**Issue by wabluk** [postsharp/Metalama#305]
When i compile my Projects i alwas get the following Exception.
Compilation completed successfully!

```
Metalama Application: Metalama.Framework
Metalama Version: 2024.1.20
Runtime: .NET Framework 4.8.9241.0
Processor Architecture: X64
OS Description: Microsoft Windows 10.0.22631 
OS Architecture: X64
Exception type: System.InvalidOperationException
Exception message: Der Vorgang ist aufgrund des aktuellen Zustands des Objekts ungültig.
===== Exception ===== 
System.InvalidOperationException: Der Vorgang ist aufgrund des aktuellen Zustands des Objekts ungültig.
   bei Metalama.Backstage.Telemetry.UsageReporter.StartSession(String kind)
   bei Metalama.Framework.Engine.Pipeline.SourceTransformer.InitializeServices(InitializeServicesContext context)
```

**Comment by prochan2** [comment]
Thank you for reporting the issue. I didn't succeed in reproducing the issue. Could you please share your source code, or at least your solution (.sln) and project (.csproj) files? And `packages.config`, if any.

How do you build your project? Which version of tooling (IDE, SDK) do you use? If you use a command line, could you please share the command?

**Comment by wabluk** [comment]
I have about 800 Project-Files and many Solutions so i can't share my Code.

I compile it with msbuild.exe
`msbuild.exe /p:Configuration=Release /p:BuildProjectReferences=false`

Here are my Packages (Directory.packages.props):
``` xml
<?xml version="1.0" encoding="utf-8"?>
<Project>
  <!-- ================================================================== -->
  <!-- Zentrale Paketverwaltung (CPM)                                     -->
  <!-- ================================================================== -->
  <PropertyGroup>
    <!-- Aktiviert zentrale Paketverwaltung (CPM) -->
    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
    <CentralPackageTransitivePinningEnabled>true</CentralPackageTransitivePinningEnabled>
    <EnablePackageVersionOverride>false</EnablePackageVersionOverride>
  </PropertyGroup>
  <ItemGroup>
    <!-- GlobalPackageReference -->
    <PackageVersion Include="log4net" Version="2.0.15" />
    <PackageVersion Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.4" />
    <PackageVersion Include="Metalama.Framework" Version="2024.1.20" />
    <PackageVersion Include="Metalama.Extensions.DependencyInjection" Version="2024.1.20" />
    <PackageVersion Include="IDisposableAnalyzers" Version="4.0.7" />
  </ItemGroup>
  <ItemGroup>
    <!-- +++ Bitte alphabetisch sortieren +++ -->
    <PackageVersion Include="Antlr" Version="3.5.0.2" />
    <PackageVersion Include="ATAPI.X64" Version="2018.3.9" />
    <PackageVersion Include="ATAPI" Version="2018.3.9" />
    <PackageVersion Include="AvalonEdit" Version="6.3.0.90" />
    <PackageVersion Include="Azure.Core" Version="1.22.0" />
    <PackageVersion Include="Azure.Storage.Common" Version="12.10.0" />
    <PackageVersion Include="Azure.Storage.Files.Shares" Version="12.9.0" />
    <PackageVersion Include="BingMapsRESTToolkit" Version="1.1.4" />
    <PackageVersion Include="bootstrap.sass" Version="4.0.0" />
    <PackageVersion Include="bootstrap" Version="4.0.0" />
    <PackageVersion Include="CsvHelper" Version="2.16.3.0" />
    <PackageVersion Include="datalogic-xamarin-sdk" Version="1.5.88" />
    <PackageVersion Include="DiffPlex.Wpf" Version="1.4.0" />
    <PackageVersion Include="docfx.console" Version="2.48.0" />
    <PackageVersion Include="DocumentFormat.OpenXml" Version="2.18.0" />
    <PackageVersion Include="DotNetZip" Version="1.16.0" />
    <PackageVersion Include="EntityFramework.de" Version="6.1.3" />
    <PackageVersion Include="EntityFramework" Version="6.1.3" />
    <PackageVersion Include="EPPlus" Version="4.5.3.3" />
    <PackageVersion Include="FluentFTP" Version="27.1.2" />
    <PackageVersion Include="ImageProcessor" Version="2.6.1.19" />
    <PackageVersion Include="iTextSharp" Version="5.5.10" />
    <PackageVersion Include="jayrock-json" Version="0.9.16530.1" />
    <PackageVersion Include="jQuery" Version="3.3.1" />
    <PackageVersion Include="jQuery.UI.Combined" Version="1.12.1" />
    <PackageVersion Include="jQuery.Validation" Version="1.17.0" />
    <PackageVersion Include="JWT" Version="5.3.1" />
    <PackageVersion Include="Leisn.MarkdigToc" Version="0.1.3" />
    <PackageVersion Include="LiveCharts.Wpf" Version="0.9.7" />
    <PackageVersion Include="LiveCharts" Version="0.9.7" />
    <PackageVersion Include="MailKit" Version="3.4.2" />
    <PackageVersion Include="Markdig" Version="0.34.0" />
    <PackageVersion Include="Cyotek.Markdig.Keyboard" Version="1.0.0-alpha4" />
    <PackageVersion Include="Markdig.SyntaxHighlighting" Version="1.1.7" />
    <PackageVersion Include="MarkdigExtensions.UrlRewriter" Version="1.0.0" />
    <PackageVersion Include="MetadataExtractor" Version="2.7.2" />
    <PackageVersion Include="Microsoft.AppCenter.Analytics" Version="4.5.3" />
    <PackageVersion Include="Microsoft.AppCenter.Crashes" Version="4.5.3" />
    <PackageVersion Include="Microsoft.AppCenter" Version="4.5.3" />
    <PackageVersion Include="Microsoft.AspNet.Identity.Core" Version="2.2.1" />
    <PackageVersion Include="Microsoft.AspNet.Mvc.FixedDisplayModes" Version="5.0.0" />
    <PackageVersion Include="Microsoft.AspNet.Mvc" Version="5.2.9" />
    <PackageVersion Include="Microsoft.AspNet.Razor" Version="3.2.9" />
    <PackageVersion Include="Microsoft.AspNet.Web.Optimization" Version="1.1.3" />
    <PackageVersion Include="Microsoft.AspNet.WebApi.Client" Version="5.2.9" />
    <PackageVersion Include="Microsoft.AspNet.WebApi.Core" Version="5.2.9" />
    <PackageVersion Include="Microsoft.AspNet.WebApi.SelfHost" Version="5.2.9" />
    <PackageVersion Include="Microsoft.AspNet.WebApi.WebHost" Version="5.2.9" />
    <PackageVersion Include="Microsoft.AspNet.WebApi" Version="5.2.9" />
    <PackageVersion Include="Microsoft.AspNet.WebPages.Data" Version="3.2.9" />
    <PackageVersion Include="Microsoft.AspNet.WebPages.WebData" Version="3.2.9" />
    <PackageVersion Include="Microsoft.AspNet.WebPages" Version="3.2.9" />
    <PackageVersion Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.1.0" />
    <PackageVersion Include="Microsoft.AspNetCore.Http.Features" Version="2.1.0" />
    <PackageVersion Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
    <PackageVersion Include="Microsoft.Azure.WebJobs.Extensions.OpenApi" Version="1.0.0" />
    <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="1.1.1" />
    <PackageVersion Include="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" Version="3.6.0" />
    <PackageVersion Include="Microsoft.Extensions.Primitives" Version="2.2.0" />
    <PackageVersion Include="Microsoft.Identity.Client" Version="4.45.0" />
    <PackageVersion Include="Microsoft.jQuery.Unobtrusive.Ajax" Version="3.2.2" />
    <PackageVersion Include="Microsoft.NET.Sdk.Functions" Version="4.1.0" />
    <PackageVersion Include="Microsoft.Office.Interop.Excel" Version="15.0.4795.1001" />
    <PackageVersion Include="Microsoft.Office.Interop.Word" Version="15.0.4797.1004" />
    <PackageVersion Include="Microsoft.Web.Infrastructure" Version="1.0.0.0" />
    <PackageVersion Include="Microsoft.Web.Services3" Version="3.0.0.0" />
    <PackageVersion Include="Microsoft.Web.WebView2" Version="1.0.2210.55" />
    <PackageVersion Include="MimeKit" Version="3.4.2" />
    <PackageVersion Include="modernhttpclient" Version="2.4.2" />
    <PackageVersion Include="ncalc" Version="1.3.8" />
    <PackageVersion Include="Newtonsoft.Json" Version="13.0.1" />
    <PackageVersion Include="NI.Email.Mime" Version="1.8.1.0" />
    <PackageVersion Include="nQuant" Version="1.0.3" />
    <PackageVersion Include="NUnit" Version="3.13.3" />
    <PackageVersion Include="NUnit3TestAdapter" Version="4.5.0" />
    <PackageVersion Include="OpenHtmlToPdf" Version="1.12.0.0" />
    <PackageVersion Include="OpenXmlPowerTools" Version="4.5.3.2" />
    <PackageVersion Include="popper.js" Version="1.14.3" />
    <PackageVersion Include="QRCoder.Unity" Version="1.0.2" />
    <PackageVersion Include="QRCoder.Xaml" Version="1.4.3" />
    <PackageVersion Include="QRCoder" Version="1.4.3" />
    <PackageVersion Include="RazorEngine" Version="3.9.3" />
    <PackageVersion Include="Rg.Plugins.Popup" Version="2.1.0" />
    <PackageVersion Include="RhinoMocks" Version="3.6.1" />
    <PackageVersion Include="RtfPipe" Version="2.0.7677.4303" />
    <PackageVersion Include="SharpZipLib" Version="1.4.1" />
    <PackageVersion Include="sqlite-net-pcl" Version="1.8.116" />
    <PackageVersion Include="SkiaSharp" Version="2.88.6" />
    <PackageVersion Include="System.Buffers" Version="4.5.1" />
    <PackageVersion Include="System.Collections.Immutable" Version="6.0.0" />
    <PackageVersion Include="System.ComponentModel.Annotations" Version="4.7.0" />
    <PackageVersion Include="System.Data.SqlClient" Version="4.8.6" />
    <PackageVersion Include="System.Diagnostics.DiagnosticSource" Version="4.6.0" />
    <PackageVersion Include="System.Drawing.Common" Version="4.7.0" />
    <PackageVersion Include="System.IO.Hashing" Version="6.0.0" />
    <PackageVersion Include="System.IO" Version="4.3.0" />
    <PackageVersion Include="System.Memory" Version="4.5.5" />
    <PackageVersion Include="System.Memory.Data" Version="1.0.2" />
    <PackageVersion Include="System.Net.Http" Version="4.3.4" />
    <PackageVersion Include="System.Numerics.Vectors" Version="4.5.0" />
    <PackageVersion Include="System.Runtime" Version="4.3.0" />
    <PackageVersion Include="System.Runtime.CompilerServices.Unsafe" Version="6.0.0" />
    <PackageVersion Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.3.0" />
    <PackageVersion Include="System.Security.Cryptography.Algorithms" Version="4.3.0" />
    <PackageVersion Include="System.Security.Cryptography.Encoding" Version="4.3.0" />
    <PackageVersion Include="System.Security.Cryptography.Primitives" Version="4.3.0" />
    <PackageVersion Include="System.Security.Cryptography.X509Certificates" Version="4.3.0" />
    <PackageVersion Include="System.Text.Encoding.CodePages" Version="6.0.0" />
    <PackageVersion Include="System.Text.Encodings.Web" Version="4.7.2" />
    <PackageVersion Include="System.Text.Json" Version="4.7.2" />
    <PackageVersion Include="System.ValueTuple" Version="4.5.0" />
    <PackageVersion Include="System.Threading.Tasks.Extensions" Version="4.5.4" />
    <PackageVersion Include="System.Windows.Interactivity.WPF" Version="2.0.20525" />
    <PackageVersion Include="WebGrease" Version="1.6.0" />    <!--Für Web.GoogleMaps-->
    <PackageVersion Include="Winnovative.PdfPrint" Version="9.0.0" />
    <PackageVersion Include="WinSCP" Version="5.9.1" />
    <PackageVersion Include="Xam.Plugin.Badge" Version="1.0.0" />
    <PackageVersion Include="Xam.Plugins.Forms.ImageCircle" Version="3.0.0.5" />
    <PackageVersion Include="Xamarin.Android.Support.Design" Version="28.0.0.3" />
    <PackageVersion Include="Xamarin.Android.Support.v7.AppCompat" Version="28.0.0.3" />
    <PackageVersion Include="Xamarin.Essentials" Version="1.7.7" />
    <PackageVersion Include="Xamarin.FFImageLoading.Forms" Version="2.4.11.982" />
    <PackageVersion Include="Xamarin.FFImageLoading.Svg.Forms" Version="2.4.11.982" />
    <PackageVersion Include="Xamarin.FFImageLoading.Transformations" Version="2.4.11.982" />
    <PackageVersion Include="Xamarin.Forms" Version="5.0.0.2578" />
    <PackageVersion Include="XmlSchemaClassGenerator-beta" Version="2.0.436" />
    <PackageVersion Include="XmpCore" Version="6.1.10.1" />
  </ItemGroup>
</Project>
```

**Comment by prochan2** [comment]
Thank you. We're going to fix this in the upcoming version. In the meantime, you can downgrade to Metalama 2024.1.19 to avoid being disturbed by this bug.

**Comment by prochan2** [comment]
This bug has been fixed in Metalama 2024.0.15, 2024.1.21, and 2024.2.8-preview.
